torch/_decomp/decompositions.py

```diff
@@ -600,6 +600,63 @@ def _log_softmax_backward_data(
     return grad_input
 
 
+@register_decomposition(aten.im2col)
+def im2col(
+    input: Tensor,
+    kernel_size: List[int],
+    dilation: List[int],
+    padding: List[int],
+    stride: List[int],
+) -> Tensor:
+
+    batch_dim = input.size(0)
```

Comment:
should you do some error checking here, like check that input dimension is 4d, kernel_size etc contain 2 elements?

Reply:
Done.

Reply:
Final nit, use _prims_common.check for error checking https://github.com/pytorch/pytorch/blob/8cd296f6804727899b39198d1641055b64f99056/torch/_prims_common/__init__.py#L1408